### DISCUSSION TITLE
Where do you set the 404 redirection?

### DISCUSSION THREAD

**Discussion by ferrarafer** [satnaing/astro-paper#47] [Q&A]
I want to know where do you set 404 redirection because I'm trying to add multi language support to my blog.

**Comment by satnaing** [comment]
We don't have to redirect any routes for 404. 
All the unknown (unspecified) pages will be redirected to `src/pages/404.astro` file.

For some edge cases, you can create an [API endpoint](https://docs.astro.build/en/core-concepts/endpoints/) and use Astro's [redirect](https://docs.astro.build/en/core-concepts/endpoints/#redirects) utility

**Comment by ferrarafer** [comment]
Hi, thanks for the quick reply ❤️ .

> All the unknown (unspecified) pages will be redirected to src/pages/404.astro file.

Where does it configured?

**Reply by satnaing** [reply]
As I stated 

> We don't have to redirect any routes for 404.

I'm not quite sure what you want to configure. 
If you want to redirect using [Astro.redirect()](https://docs.astro.build/en/guides/server-side-rendering/#astroredirect) method, you need to enable SSR mode since Astro.redirect() isn't [available in static mode](https://docs.astro.build/en/reference/errors/static-redirect-not-available/).

Or you can use [meta refresh attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta) in static mode as [described](https://docs.astro.build/en/reference/errors/static-redirect-not-available/#:~:text=To%20redirect%20on%20a%20static%20website%2C%20the%20meta%20refresh%20attribute%20can%20be%20used.) in Astro's docs.

If you're using a third-party packages like [astro-i18next](https://github.com/yassinedoghri/astro-i18next), you might need to [configure routes in its config file](https://github.com/yassinedoghri/astro-i18next#-translate-routes).
